libraries/AP_Math/AP_Math_PX4.h

```diff
@@ -0,0 +1,159 @@
+#pragma once
+#undef isfinite
+
+#include <limits>
+#include <type_traits>
+#include <cmath>
+
+#include <math.h>
+#include <stdint.h>
+
+#include "AP_Math_Common.h"
+
+
+namespace std {
+    template<class T>
+    bool isfinite(const T& val) {
```

Comment:
we can move this later when the PR gets applied, but have you seen that now we have a way to workaround broken headers from toolchains? We started to use wrapper headers. See `libraries/AP_Common/missing/`. When we finish the conversion of the PX4 build system is finished the idea is to check these things during the configure phase.